From c7be682ca37fb9b330788824e87b159ae247f8f5 Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Fri, 30 Jul 2021 10:26:25 +0800
Subject: [PATCH] ofpwq: assert ofp_max_linkhdr with static

When build opd package, there is below error:
multiple definition of `ofp_max_linkhdr'
Because ofp_max_linkhdr is only used in local static
function, add static assertion to it.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 src/ofp_tcp_subr.c   | 2 +-
 src/ofp_udp_usrreq.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ofp_tcp_subr.c b/src/ofp_tcp_subr.c
index c68198c..7c19945 100644
--- a/src/ofp_tcp_subr.c
+++ b/src/ofp_tcp_subr.c
@@ -77,7 +77,7 @@
 #define	SYSCTL_VNET_INT OFP_SYSCTL_INT
 
 unsigned int ofp_max_protohdr = 0;
-int ofp_max_linkhdr = 64;
+static int ofp_max_linkhdr = 64;
 
 VNET_DEFINE(int, ofp_tcp_mssdflt) = OFP_TCP_MSS;
 #ifdef INET6
diff --git a/src/ofp_udp_usrreq.c b/src/ofp_udp_usrreq.c
index 2f62607..373c290 100644
--- a/src/ofp_udp_usrreq.c
+++ b/src/ofp_udp_usrreq.c
@@ -95,7 +95,7 @@ int		ofp_udp_log_in_vain = 0;
 int		ofp_udp_blackhole = 0;
 uint64_t	ofp_udp_sendspace = 9216;		/* really max datagram size */
 uint64_t	ofp_udp_recvspace = 40 * (1024 + sizeof(struct ofp_sockaddr_in6));
-int		ofp_max_linkhdr;
+static int		ofp_max_linkhdr;
 VNET_DEFINE(int, ofp_ip_defttl) = 255;
 
 struct inpcbhead ofp_udb;		/* from udp_var.h */
-- 
2.17.1

